client/src/app/components/presentation/common/modal/UploadModal.jsx

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {imageUploadToCloud, imageUploadToDb} from '../../../../actions/uploadImage';
+import ShowProgressBar from '../Preloader/ShowProgressBar.jsx'
+
+/**
+ * display modal with a file input field and a submit button
+ * @class UploadPictureModal
+ * @extends CommonModal
+ */
+class UploadModal extends React.Component {
+  /**
+   * @constructor
+   * @extends React.Component
+   * @param {object} props
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      isLoading: false
+      
+
+
+    };
+    this.triggerFileSelect = this
+      .triggerFileSelect
+      .bind(this);
+    this.onInputChange = this
+      .onInputChange
+      .bind(this);
+      this.onClick = this
+      .onClick
+      .bind(this);
+  }
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.username && nextProps.username !== ''){
+      this.setState({username: nextProps.username})
+    }
+  }
+  /**
+   * handle change in file input
+   * @method onInputChange
+   * @memberof UploadPictureModal
+   * @param {object} event
+   * @return {void}
+   */
+  onInputChange(event) {
+    event.preventDefault();
+    this.setState({isLoading: true});
+    // let reader = new FileReader();
+    const profilePic = event.target.files[0];
+
+    const fileExt = UploadModal.getFileExtension(event.target.files[0].name);
+    const filename = `${this.props.username}.${fileExt}` || event.target.files[0].name;
+
+  
+    this
+      .props
+      .imageUploadToCloud(this.props.username, profilePic)
+      .then((response)=>{
+        this.setState({isLoading: false})
+          console.log('this should happen', response)
+          this.setState({filename});
+          
+      }
+
+      )
+      
+  }
+
+
+
+  onClick(event){
+    event.preventDefault();
+    this.setState({isLoading: true});
+    if(this.state.filename)
+      {
+        console.log('I am here', this.props.secure_url)
+        // imageUploadToDb
```

Comment:
Zombie code. I will like you to replicate killing this wherever you have such existing.